src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -0,0 +1,218 @@
+                        case 'L': return TryGet(AcceptLanguage, key, startIndex, length, out name); // Accept-[L]anguage
+                    }
+                    break;
+
+                case 16:
+                    switch (key[startIndex + 11])
+                    {
+                        case 'o': return TryGet(ContentEncoding, key, startIndex, length, out name); // Content-Enc[o]ding
+                        case 'g': return TryGet(ContentLanguage, key, startIndex, length, out name); // Content-Lan[g]uage
+                        case 'a': return TryGet(ContentLocation, key, startIndex, length, out name); // Content-Loc[a]tion
+                        case 'c': return TryGet(ProxyConnection, key, startIndex, length, out name); // Proxy-Conne[c]tion
+                        case 'i': return TryGet(WWWAuthenticate, key, startIndex, length, out name); // WWW-Authent[i]cate
+                        case 'r': return TryGet(XAspNetVersion, key, startIndex, length, out name);  // X-AspNet-Ve[r]sion
+                    }
+                    break;
+
+                case 17:
+                    switch (key[startIndex])
+                    {
+                        case 'I': return TryGet(IfModifiedSince, key, startIndex, length, out name);  // [I]f-Modified-Since
+                        case 'S': return TryGet(SecWebSocketKey, key, startIndex, length, out name);  // [S]ec-WebSocket-Key
+                        case 'T': return TryGet(TransferEncoding, key, startIndex, length, out name); // [T]ransfer-Encoding
+                    }
+                    break;
+
+                case 18:
+                    return TryGet(ProxyAuthenticate, key, startIndex, length, out name); // Proxy-Authenticate
+
+                case 19:
+                    switch (key[startIndex])
+                    {
+                        case 'C': return TryGet(ContentDisposition, key, startIndex, length, out name); // [C]ontent-Disposition
+                        case 'I': return TryGet(IfUnmodifiedSince, key, startIndex, length, out name);  // [I]f-Unmodified-Since
+                        case 'P': return TryGet(ProxyAuthorization, key, startIndex, length, out name); // [P]roxy-Authorization
+                    }
+                    break;
+
+                case 20:
+                    return TryGet(SecWebSocketAccept, key, startIndex, length, out name); // Sec-WebSocket-Accept
+
+                case 21:
+                    return TryGet(SecWebSocketVersion, key, startIndex, length, out name); // Sec-WebSocket-Version
+
+                case 22:
+                    return TryGet(SecWebSocketProtocol, key, startIndex, length, out name); // Sec-WebSocket-Protocol
+
+                case 24:
+                    return TryGet(SecWebSocketExtensions, key, startIndex, length, out name); // Sec-WebSocket-Extensions
+            }
+
+            // When adding a new constant, add it to HttpKnownHeaderNames.cs as well.
```

Comment:
Yeah, I intentionally added the same comment at both the top and bottom of the file. Based on your feedback, I went ahead and removed the comment at the bottom. One comment at the top should be sufficient.